### PR TITLE
feat: disable auto-assigned activities/flows from assignment (M2-7604)

### PR DESCRIPTION
**Pull request by LashaunnaS**
- [x] Tests for the changes have been added
- [x] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-7604](https://mindlogger.atlassian.net/browse/M2-7604)

Within our `enableActivityAssign` feature flag, as an admin you should be able to know why activities or flows are disabled in the Assign flow.

Changes include:
- In the Assign Activity panel, from the Select Activities list, **auto-assigned activities/flows are displayed as disabled and can not be selected by default.**
- **A black tooltip appears when hovering over each disabled activity/flow** to provide information on why the activity/flow is disabled.
  - Activity copy: This activity is set to ‘Auto-assign’ in the Activity Builder and therefore cannot be assigned here.
  - Flow copy: This flow is set to ‘Auto-assign’ in the Activity Flow Builder and therefore cannot be assigned here.
  - Note: Added FR translations from google.
- If all of the activities and/or flows in the assign flow are marked as “auto assigned” and therefore d**isabled, the “select all” button should also be disabled.**

### 📸 Screenshots

| Before (Optional)                      | After                                 |
| -------------------------------------- | ------------------------------------- |
| https://www.loom.com/share/6099214544384f0a8216d548fe8737f3?sid=62ffb3ac-c31b-4e34-8cf5-b11cee2aa3bc | https://www.loom.com/share/22464dd4f70d4a29b23cb9fbf3ba49c5?sid=95f7ff5d-bbc6-4c90-9aa0-c962135f192f |

#### All options disabled:
**Test steps**...
- Create or update two Activities so that `Auto-assign this activity (as self-report)` is checked on both.
- Do the same for flows.
- Navigate to the applet activities tab containing those activities and flows.
- Click on "Assign"

**Expected outcome...**
- "Select All" should be disabled and a light grey.
- All activities and flows should also be disabled and a lighter grey.
- When hovering over an item, you should see a tooltip appear on the left with further details.
  - Changing the lang to 'fr' should update the tooltip text. 

#### Some options disabled:
**Test steps**...
- Update one of the Activities so that `Auto-assign this activity (as self-report)` is **unchecked**
- Do the same for flows.
- Navigate to the applet activities tab containing those activities and flows.
- Click on "Assign"

**Expected outcome...**
- "Select All" should be enabled.
- Clicking on select all should only select the activities / flows that are not marked "auto assign".
  - the summary note aligned with the select all button should note the correct amount of selected items.
- The right activities and flows should be disabled and a lighter grey while the others are active and clickable.
- When hovering over an active item, you should **not** see a tooltip appear on the left with further details.

### ✏️ Notes

- These feature changes are wrapped in the `enableActivityAssign` flag.

### CODE REVIEW

File: src/modules/Dashboard/components/ActivityAssignDrawer/ActivityCheckbox/ActivityCheckbox.tsx
Comment (LashaunnaS) [inline src/modules/Dashboard/components/ActivityAssignDrawer/ActivityCheckbox/ActivityCheckbox.tsx]: Left a figma [comment](https://www.figma.com/design/6BNx63peNVzrHAF1FoZ78x?node-id=10933-129273#957753165) for Matt to confirm this colour for disabled.

File: src/modules/Dashboard/components/ActivityAssignDrawer/ActivitiesList/ActivitiesList.tsx
Comment (mbanting) [inline src/modules/Dashboard/components/ActivityAssignDrawer/ActivitiesList/ActivitiesList.tsx]: I don't think the feature flag check is needed since the parent component that uses this (`ActivityAssignDrawer`) is effectively behind the `enableActivityAssign` feature flag wherever it's used.

File: src/modules/Dashboard/components/ActivityAssignDrawer/ActivityAssignDrawer.tsx
Comment (mbanting) [inline src/modules/Dashboard/components/ActivityAssignDrawer/ActivityAssignDrawer.tsx]: Same thing here. It's safe to assume `ActivityAssignDrawer` isn't accessible if `enableActivityAssign` is not enabled.

**Review by mbanting — CHANGES_REQUESTED** [review]
Great job @LashaunnaS ! Thank you for the super helpful PR description. I know it's still in draft mode but thought I'd give some quick feedback already. Left a couple comments about the feature flag.

Also for the PR title, can you adjust it the following format
`type(optional scope): brief summary (M2-####[, additional tickets])`

ie: `feat: disable auto-assigned activities/flows from assignment (M2-7604)`

- ok to base the title on the JIRA ticket title but feel free to tweak it if necessary and drop the [BE][FE][etc] prefixes
- super helpful to include the ticket number

Doing the above makes both the git history and [release notes](https://github.com/ChildMindInstitute/mindlogger-admin/releases/tag/v1.39.0) even more useful.
 
(more info [here](https://mindlogger.atlassian.net/wiki/spaces/MINDLOGGER1/pages/7503873/Configuration+Management+Plan#Pull-Requests))

Also, reminder that we've adopted a new workflow for web and admin tickets (like this one). Once you have the approvals, don't merge yet. Instead update the ticket to `To Be Tested` for QA to pickup and test in the [feature environment](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1926#issuecomment-2364568935).

Thanks!

**Review by farmerpaul — APPROVED** [review]
Excellent first PR @LashaunnaS – can't see anything to provide feedback on, other than great work!

File: src/shared/styles/variables/palette.ts
Comment (LashaunnaS) [inline src/shared/styles/variables/palette.ts:36]: @farmerpaul is there a process to how we go about adding colours? This colour is to address the disabled state on the checkbox but I wasn't sure if we have some method behind the pallet colours. 🤔

Comment (farmerpaul) [reply src/shared/styles/variables/palette.ts:36]: That's what I'd do. There's no system in place of syncing tokens from Figma to the code. Just whenever you see a missing one, add it just like you've done here! :+1:

**Comment by LashaunnaS** [comment]
@egodoy-metalab new feature

**Review by mbanting — APPROVED** [review]
Amazing @LashaunnaS !

**Pull request merged**
